pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -485,6 +536,46 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Set routers for a particular domain.
+		#[pallet::weight(T::WeightInfo::set_domain_multi_router())]
+		#[pallet::call_index(9)]
+		pub fn set_domain_multi_router(
+			origin: OriginFor<T>,
+			domain: Domain,
+			routers: BoundedVec<T::Router, T::MultiRouterCount>,
+		) -> DispatchResult {
+			T::AdminOrigin::ensure_origin(origin)?;
+
+			ensure!(domain != Domain::Centrifuge, Error::<T>::DomainNotSupported);
+			ensure!(
+				routers.len() == T::MultiRouterCount::get() as usize,
+				Error::<T>::InvalidMultiRouter
+			);
+
+			let mut router_hashes = Vec::new();
+
+			for router in &routers {
+				router.init().map_err(|_| Error::<T>::RouterInitFailed)?;
```

Comment:
Awesome suggestion. thank you!

Reply:
I didn't know substrate supported custom content in the error; if it works, amazing!

Reply:
Substrate did not support this from Genesis but IIRC it was added early last year. However, since we have lacked behind Substrate versions so much, we haven't utilised it yet.

Reply:
It seems we can add other `PalletError`s in there - https://github.com/paritytech/polkadot-sdk/blob/149c70938f2b29f8d92ba1cc952aeb63d4084e27/bridges/snowbridge/pallets/inbound-queue/src/lib.rs#L185.